yaml/shared/src/main/scala/org/virtuslab/yaml/syntax/YamlPrimitive.scala

```diff
@@ -0,0 +1,23 @@
+package org.virtuslab.yaml.syntax
+
+import org.virtuslab.yaml.Node
+import org.virtuslab.yaml.Node.ScalarNode
+
+import scala.language.implicitConversions
+
+final case class YamlPrimitive(node: Node)
+object YamlPrimitive:
+  given Conversion[String, YamlPrimitive]  = s => YamlPrimitive(ScalarNode(s))
+  given Conversion[Boolean, YamlPrimitive] = s => YamlPrimitive(ScalarNode(s.toString))
+  given Conversion[Int, YamlPrimitive]     = s => YamlPrimitive(ScalarNode(s.toString))
+  given Conversion[Long, YamlPrimitive]    = s => YamlPrimitive(ScalarNode(s.toString))
+  given Conversion[Node, YamlPrimitive]    = s => YamlPrimitive(s)
+
+  given conv[A, B](using
+      Conversion[A, YamlPrimitive],
+      Conversion[B, YamlPrimitive]
+  ): Conversion[(A, B), (YamlPrimitive, YamlPrimitive)] = (s, b) => (s, b)
+
+  given convMulti[A, B](using
+      Conversion[A, YamlPrimitive]
+  ): Conversion[(A, Node), (YamlPrimitive, YamlPrimitive)] = (s, b) => (s, YamlPrimitive(b))
```

Comment:
I think we could do it in more explicit way
```suggestion
  given [A, B](using
      c1: Conversion[A, YamlPrimitive],
      c2: Conversion[B, YamlPrimitive]
  ): Conversion[(A, B), (YamlPrimitive, YamlPrimitive)] = (a, b) => (c1(a), c2(b))

  given [A](using
      c: Conversion[A, YamlPrimitive]
  ): Conversion[(A, Node), (YamlPrimitive, YamlPrimitive)] = (a, node) => (c(a), YamlPrimitive(node))
```